components/TextFields/src/ContainedInputView/MDCBaseTextFieldLayout.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@interface MDCBaseTextFieldLayout : NSObject
+
+@property(nonatomic, assign) BOOL leftViewHidden;
+@property(nonatomic, assign) BOOL rightViewHidden;
+
+@property(nonatomic, assign) CGRect leftViewFrame;
+@property(nonatomic, assign) CGRect rightViewFrame;
+
+- (nonnull instancetype)initWithTextFieldSize:(CGSize)textFieldSize
+                                     leftView:(nullable UIView *)leftView
+                                 leftViewMode:(UITextFieldViewMode)leftViewMode
+                                    rightView:(nullable UIView *)rightView
+                                rightViewMode:(UITextFieldViewMode)rightViewMode
```

Comment:
Should these arguments be adjusted to leading/trailing to match the public APIs?

Reply:
This was actually intentional. I figured the layout object shouldn't have to care about the locale-dependent notion of leading and trailing. It's only concerned about where the things go within the bounds of the view--on the left or right side.

Reply:
This was actually intentional. I figured the layout object shouldn't have to care about the locale-dependent notion of leading and trailing. It's only concerned about where the things go within the bounds of the view--on the left or right side.

Reply:
How does the isRTL property affect which of these properties are used during layout?

The complexity here likely warrants some additional documentation, or alternatively I think the reader may find this code easier to follow if we consistently apply the leading/trailing semantics as we've done in other components. We can address this in a follow-up though.

Reply:
Whoops. Come to think of it, it doesn't! It should've been brought in later I guess. The layout object _does_ need to know if it's RTL for other things, like the clear button and label (both floating and non-floating) positions, for example.